SAM2_Tracking/utils.py

```diff
@@ -35,14 +35,14 @@ def read_config_yaml(config_path):
 
     return config
 
-def adjust_annotations(annotations_file=None, fps=None, SAM2_start=None, 
+def adjust_annotations(annotations_file=None, fps=None, out_fps=None, SAM2_start=None, 
                        df_columns=None, frame_col_name=None):
     """
     Reads in a dictionary of annotations and converts it to a Pandas 
     DataFrame. Additionally, adjusts provided annotations so the 
     frame value aligns with SAM2. Adjustment is dictated by the 
-    formula: `(frame_number - SAM2_start - 1) / (fps / 3)`, which 
-    is then rounded and turned into an integer. 
+    formula: `(frame_number * (fps / out_fps) + SAM2_start, which 
```

Comment:
```suggestion
    formula: `frame_number * (fps / out_fps) + SAM2_start, which 
```
I removed the `(`. I am also curious about this change in the formula. I might be wrong, but I do not think `(frame_number - SAM2_start) / (fps / 3)` is equal to `frame_number * (fps / out_fps) + SAM2_start`

Reply:
I do see that this new formula matches the one we have in `write_output_video`, so it must be the one we want?

Reply:
Good catch! I was investigating this formula rigorously, and also concerned by it's previous mismatch from the write-output video, and it seems I didn't incorporate all the changes here. I have now changed this formula to (frame_index - SAM2_start) / (fps / out_fps), and adjusted the actual formula in the code to match. This is very similar to what it was before, but I removed the `-1` from the original formula because the annotation `frame_number`, or `frame_index` begins counting at 0, and the list of frames that are ingested by SAM2 also begins counting at 0. Previously, the actual image labels ingested by SAM2 were used later to reference the provided images to annotation frames, and those image labels begin counting at 0001. So with the latest workflow, the subtraction of 1 is no longer necessary. 

The updated formula is the inverse of what is found in the write_output_video because the formula here to adjust annotations takes the original annotation frame numbers and converts them to the extracted SAM2 frame numbers, while the write_output_video formula takes the extracted SAM2 frame numbers and converts them back to the original annotation frame numbers for easy correcting.

Reply:
Awesome, thank you for the detailed explanation and for fixing that!